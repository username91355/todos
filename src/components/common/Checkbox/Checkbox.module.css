.checkbox {
    padding: 10px;
}

.checkbox__wrapper {
    display: flex;
    align-items: center;
    box-sizing: border-box;
    text-align: center;
    cursor: pointer;
}

.checkbox__input {
    appearance: none;
    position: relative;
    padding-left: 22px;
    cursor: pointer;
}

.checkbox__input_not_select:before {
    content: '';
    top: calc(50% - 8px);
    left: calc(50% - 8px);
    position: absolute;
    height: 20px;
    width: 20px;
    background-color: white;
    border-radius: 5px;
}

.checkbox__input_select:before {
    content: '\2713';
    font-size: 20px;
    font-weight: 900;
    color: green;
    top: calc(50% - 8px);
    left: calc(50% - 8px);
    position: absolute;
    height: 20px;
    width: 20px;
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: white;
    border-radius: 5px;
}

.checkbox__span {
    color: white;
    cursor: pointer;
    line-height: 1em;
}