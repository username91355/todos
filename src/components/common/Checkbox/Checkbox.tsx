import React, {ChangeEvent} from 'react';
import s from './Checkbox.module.css';

interface IProps extends React.HTMLProps<HTMLInputElement> {
    checked: boolean
    onChangeChecked?: (value: boolean) => void
    spanClassName?: string
}

export const Checkbox: React.FC<IProps> = React.memo(props => {

    const {
        checked,
        onChange,
        onChangeChecked,
        spanClassName,
        label,
        ...restProps
    } = props;

    const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
        onChange && onChange(e);
        onChangeChecked && onChangeChecked(e.currentTarget.checked);
    };

    const isSelectClass = checked ? s.checkbox__input_select : s.checkbox__input_not_select;

    return (
        <div className={s.checkbox}>
            <label className={s.checkbox__wrapper}>
                <input
                    type={'checkbox'}
                    checked={checked}
                    onChange={onChangeHandler}
                    className={`${s.checkbox__input} ${isSelectClass}`}
                    {...restProps}
                />
                <span
                    className={`${s.checkbox__span} ${spanClassName}`}
                >
                    {label}
                </span>
            </label>
        </div>
    );
});