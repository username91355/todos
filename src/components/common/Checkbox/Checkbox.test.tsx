import {render, screen} from '@testing-library/react';
import {Checkbox} from './Checkbox';
import userEvent from '@testing-library/user-event';
import {ChangeEvent} from 'react';

let onChange: (e: ChangeEvent<HTMLInputElement>) => void;
let onChangeChecked: (checked: boolean) => void;

beforeEach(()=> {
    onChange = jest.fn();
    onChangeChecked = jest.fn();
});

describe('Button test`s', () => {

    it('should be render checked true', () => {
        render(<Checkbox
            checked={true}
            onChange={onChange}
        />);
        expect(screen.getByRole('checkbox')).toBeInTheDocument();
        expect(screen.getByRole('checkbox')).toBeChecked();
    });

    it('should be render checked false', () => {
        render(<Checkbox
            checked={false}
            onChange={onChange}
        />);
        expect(screen.getByRole('checkbox')).toBeInTheDocument();
        expect(screen.getByRole('checkbox')).not.toBeChecked();
    });

    it('should be call onChange and onChangeChecked', () => {
        render(<Checkbox
            checked={false}
            onChange={onChange}
            onChangeChecked={onChangeChecked}
        />);
        userEvent.click(screen.getByRole('checkbox'));
        expect(screen.getByRole('checkbox')).toBeInTheDocument();
        expect(onChange).toBeCalledTimes(1);
        expect(onChangeChecked).toBeCalledTimes(1);
    });

    it('checkbox snapshot', () => {
        const view = render(<Checkbox
            checked={true}
        />);
        expect(view).toMatchSnapshot();
    });
});