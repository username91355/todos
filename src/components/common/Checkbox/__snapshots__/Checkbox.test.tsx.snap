// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test\`s checkbox snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="checkbox"
      >
        <label
          class="checkbox__wrapper"
        >
          <input
            checked=""
            class="checkbox__input checkbox__input_select"
            type="checkbox"
          />
          <span
            class="checkbox__span undefined"
          />
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="checkbox"
    >
      <label
        class="checkbox__wrapper"
      >
        <input
          checked=""
          class="checkbox__input checkbox__input_select"
          type="checkbox"
        />
        <span
          class="checkbox__span undefined"
        />
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
